tests/benchmark/RuntimeBench.cpp

```diff
@@ -0,0 +1,612 @@
+  template <typename BackendTy>                                                \
+  class name##component##Benchmark : public component##Benchmark<BackendTy> {  \
+  protected:                                                                   \
+    void setUpModule(benchmark::State &state) override {                       \
+      this->mod_ = moduleCreator();                                            \
+    }                                                                          \
+  };
+
+/// Declare subclasses of all RuntimeBenchmark subclasses with appropriate
+/// overrides.
+#define DECLARE_RUNTIME_BENCHMARK(name, moduleCreator, dagCreator)             \
+  DECLARE_RUNTIME_COMPONENT_BENCHMARK(name, moduleCreator, HostManager)        \
+  DECLARE_EXECUTOR_BENCHMARK(name, moduleCreator, dagCreator)                  \
+  DECLARE_RUNTIME_COMPONENT_BENCHMARK(name, moduleCreator, DeviceManager)
+
+/// Define a RuntimeBenchmark subclass declared using
+/// DECLARE_XXX_BENCHMARK for a specific backend and component. This instance
+/// calls RuntimeBenchmark::runBenchmark to run the benchmark.
+#define INSTANTIATE_RUNTIME_COMPONENT_BENCHMARK(name, backend, component)      \
+  BENCHMARK_TEMPLATE_DEFINE_F(name##component##Benchmark, component##backend,  \
+                              backend)                                         \
+  (benchmark::State & state) { runBenchmark(state); }                          \
+  BENCHMARK_REGISTER_F(name##component##Benchmark, component##backend)         \
+      ->Unit(benchmark::kMicrosecond);
+
+/// Define RuntimeBenchmark subclasses for all runtime components.
+#define INSTANTIATE_RUNTIME_BENCHMARK(name, backend)                           \
+  INSTANTIATE_RUNTIME_COMPONENT_BENCHMARK(name, backend, HostManager)          \
+  INSTANTIATE_RUNTIME_COMPONENT_BENCHMARK(name, backend, Executor)             \
+  INSTANTIATE_RUNTIME_COMPONENT_BENCHMARK(name, backend, DeviceManager)
+
+//===--------------------------------------------------------------------===//
+//                       Common Utility Functions                           //
+//===--------------------------------------------------------------------===//
+
+/// Set up a DeviceManager instance by instantiating it, compiling all the
+/// functions in \p mod using \p backend and adding them to the DeviceManager
+/// instance. A reference to the configured DeviceManager instance is returned
+/// in \p deviceManager, and all CompiledFunctions creating during compilation
+/// are returned in \p deviceManagerFunctions.
+void setUpDeviceManagerCommon(
+    benchmark::State &state, std::unique_ptr<Backend> &backend,
+    std::unique_ptr<Module> &mod, std::unique_ptr<DeviceManager> &deviceManager,
+    std::unordered_map<std::string, std::unique_ptr<CompiledFunction>>
+        &deviceManagerFunctions) {
+
+  // Check that the backend is valid.
+  if (!backend) {
+    state.SkipWithError(
+        "Unable to set up device manager - backend not set up!");
```

Comment:
nit: for this and other errors: we tend to use DeviceManager over device manager.